src/fst.jl

```diff
@@ -1120,30 +1120,29 @@ for i = 1:10 body end
 
 - https://github.com/domluna/JuliaFormatter.jl/issues/34
 """
-function eq_to_in_normalization!(fst::FST, always_for_in::Bool)
+function eq_to_in_normalization!(fst::FST, always_for_in::Bool, for_in_replacement::String)
     if fst.typ === Binary
         idx = findfirst(n -> n.typ === OPERATOR, fst.nodes)
         idx === nothing && return
         op = fst[idx]
 
+        !valid_for_in_op(op.val) && return
+
         if always_for_in
-            op.val = "in"
+            op.val = for_in_replacement
             op.len = length(op.val)
-            return
-        end
-
-        if op.val == "=" && op_kind(fst[end]) !== Tokens.COLON
+        elseif op.val == "=" && op_kind(fst[end]) !== Tokens.COLON
             op.val = "in"
             op.len = length(op.val)
         elseif op.val == "in" && op_kind(fst[end]) === Tokens.COLON
             op.val = "="
             op.len = length(op.val)
         end
-    elseif fst.typ === Block || fst.typ === Brackets
-        for n in fst.nodes
-            eq_to_in_normalization!(n, always_for_in)
+    elseif fst.typ === Block || fst.typ === Brackets || fst.typ === Filter
```

Comment:
made this just Filter of !is_leaf since it's a more minimal change

Reply:
ah great, yes, that should work pretty well